src/AutorestSwift/View Models/ServiceClientViewModel.swift

```diff
@@ -42,23 +46,158 @@ struct ParameterViewModel {
+        // and add the request signature parameter to operation parameters
+        var params = [ParameterViewModel]()
+        for param in request.signatureParameters ?? [] {
+            params.append(ParameterViewModel(from: param))
+        }
+        self.params = params
+
+        self.objectType = request.signatureParameters?.first?.schema.name
+    }
+}
+
+struct ResponseViewModel {
+    let statusCodes: [String]
+    let knownMediaType: String?
+    let mediaTypes: [String]?
+    let objectType: String?
+
+    init(from response: Response) {
+        let httpResponse = response.protocol.http as? HttpResponse
+        var statusCodes = [String]()
+        httpResponse?.statusCodes.forEach { statusCodes.append($0.rawValue) }
+
+        self.statusCodes = statusCodes
+        self.knownMediaType = httpResponse?.knownMediaType?.rawValue
+        self.mediaTypes = httpResponse?.mediaTypes
+
+        // check if the request body schema type is object, store the object type of the response body
+        let schemaResponse = response as? SchemaResponse
+        self.objectType = schemaResponse?.schema.name
+    }
+}
+
 struct OperationViewModel {
     let name: String
     let comment: ViewModelComment
     let params: [ParameterViewModel]
     let returnType: ReturnTypeViewModel?
+    let queryParams: [KeyValueViewModel]?
+    let headerParams: [KeyValueViewModel]?
+    let uriParams: [KeyValueViewModel]?
+    let requests: [RequestViewModel]?
+    let responses: [ResponseViewModel]?
+    let method: String?
+    let path: String?
+    private static var logger = Logger(withName: "Autorest.Swift.OperationViewModel")
 
     init(from schema: Operation) {
         self.name = operationName(for: schema.name)
         self.comment = ViewModelComment(from: schema.description)
 
+        // operation parameters should be all the signature paramters from the schema
+        // and the signature parameters of the Request
         var items = [ParameterViewModel]()
         for param in schema.signatureParameters ?? [] {
             items.append(ParameterViewModel(from: param))
         }
+
+        var queryParams = [KeyValueViewModel]()
+        var headerParams = [KeyValueViewModel]()
+        var uriParams = [KeyValueViewModel]()
+
+        for param in schema.parameters ?? [] {
+            let httpParam = param.protocol.http as? HttpParameter?
+
+            switch httpParam??.in {
+            case .query:
+                queryParams.append(KeyValueViewModel(from: param, signatureParameters: items))
+            case .header:
+                headerParams.append(KeyValueViewModel(from: param, signatureParameters: items))
+            case .uri:
+                uriParams.append(KeyValueViewModel(from: param, signatureParameters: items))
+
+            default:
+                // TODO: - implemented
+                OperationViewModel.logger.log("Http Parameter \(httpParam??.in) is not support")
+            }
+        }
```

Comment:
```suggestion
        for param in schema.parameters ?? [] {
            guard let httpParam = param.protocol.http as? HttpParameter else { continue }

            switch httpParam.in {
            case .query:
                queryParams.append(KeyValueViewModel(from: param, signatureParameters: items))
            case .header:
                headerParams.append(KeyValueViewModel(from: param, signatureParameters: items))
            case .uri:
                uriParams.append(KeyValueViewModel(from: param, signatureParameters: items))
            default:
                continue
            }
        }
```

The `foo??.property` syntax is one I have not come across before. If you do, then there's almost certain a way to simplify it. The default can simply be to continue here since the other ParameterLocation options are presumably not rendered as key/value pairs.

Reply:
Thanks Travis.